x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/converters/internal_rule_to_api_response.ts

```diff
@@ -66,5 +67,6 @@ export const internalRuleToAPIResponse = (
     actions: [...actions, ...(systemActions ?? [])],
     // Execution summary
     execution_summary: executionSummary ?? undefined,
+    rule_source: snakecaseKeys(normalizedRuleParams.ruleSource, { deep: true }),
```

Comment:
Isn't `rule_source` already being converted by `commonParamsCamelToSnake`?

Reply:
@xcrzx It does, this was more of a typing thing as `commonParamsCamelToSnake` takes a type of `BaseRuleParams` which still declares `rule_source` as optional so it doesn't fit the expected return type in this file (`rule_source` might be undefined). My other idea was to cast with a custom type when we spread the common params into this object, that might be better since it's not actually performing logic twice

Reply:
I see, thanks for the clarification. We might need a separate type to represent normalized rule parameters where `rule_source` is required. That way, we can make `commonParamsCamelToSnake` a generic function that works with both standard rule parameters and normalized ones. 

I'll leave it up to you, both approaches work for me!